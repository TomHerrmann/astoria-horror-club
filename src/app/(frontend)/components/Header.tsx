import { Creepster } from 'next/font/google'

const creepster = Creepster({
  subsets: ['latin'],
  weight: '400',
})

/**
 *
 * @returns Global Header coponents for AHC app
 */
export default function Header() {
  return (
    <header className="flex justify-center">
      <h1 className={`text-6xl ${creepster.className}`}>Astoria Horror Club</h1>
    </header>
  )
}
