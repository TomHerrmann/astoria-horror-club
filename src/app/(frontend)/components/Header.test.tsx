import { describe, it, expect, vi } from 'vitest'
import { render, screen } from '@testing-library/react'
import Header from './Header'

// Mock the next/font/google module
vi.mock('next/font/google', () => ({
  Creepster: vi.fn().mockReturnValue({
    className: 'mock-creepster-font', // Mock className we can test for
  }),
}))

describe('Header', () => {
  it('matches snapshot', () => {
    const { container } = render(<Header />)
    expect(container).toMatchSnapshot()
  })

  it('renders the header with correct text', () => {
    render(<Header />)
    expect(screen.getByText('Astoria Horror Club')).toBeDefined()
  })
})
