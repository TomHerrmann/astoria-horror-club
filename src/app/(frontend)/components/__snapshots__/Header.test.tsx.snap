// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > matches snapshot 1`] = `
<div>
  <header
    class="flex justify-center"
  >
    <h1
      class="text-6xl mock-creepster-font"
    >
      Astoria Horror Club
    </h1>
  </header>
</div>
`;
